src/components/ProfilePage.jsx

```diff
@@ -1,81 +1,147 @@
-  profileCard
-} from '../styles';
+import { submitButtonStyle, loginFormContainerStyle } from '../styles';
 import { UserConsumer } from '../contexts';
 import { profilePageValidator } from '../helpers';
+import imageUrl from '../images/imageURL.jpg';
+import noImageUrl from '../images/noImageUrl.png';
 
 const ProfilePage = props => {
+  const [editStatus, setEditStatus] = useState(false);
+  const [editButtonName, seteditButtonName] = useState('Edit Profile');
+
+  const handleOnEdit = () => {
+    if (editButtonName === 'Edit Profile') {
+      seteditButtonName('Finished');
+      setEditStatus(!editStatus);
+    } else {
+      seteditButtonName('Edit Profile');
+      setEditStatus(!editStatus);
+    }
+  };
+
   return (
-    <div className="card" style={profileCard}>
+    <div>
       <UserConsumer>
         {({ user, handleUserData }) => (
-          <Formik
-            initialValues={{ username: user.username, email: user.email }}
-            validate={values => profilePageValidator(values)}
-            onSubmit={async ({ username, email }, { setSubmitting }) => {
-              updateUserInfo({ username, email, id: props.match.params.userId })
-                .then(res => {
-                  handleUserData({ username, email });
-                  swal({
-                    type: 'success',
-                    position: 'top-end',
-                    title: 'Update Successful',
-                    toast: true,
-                    showConfirmButton: false,
-                    timer: 3000
-                  });
+          <div className="card profileCard">
+            <div className="profileImgBoxBackground">
+              <div className="editButton">
+                <button
+                  type="button"
+                  onClick={() => handleOnEdit()}
+                  className="btn btn-primary"
+                >
+                  {editButtonName}
+                </button>
+              </div>
+              <div className="profileImgBox">
+                {user.imageURL && (
+                  <img className="profileImg" src={imageUrl} alt="" />
+                )}
+                {!user.imageURL && (
+                  <img className="profileImg" src={noImageUrl} alt="" />
+                )}
+              </div>
+            </div>
+
+            <div className="profileDetails">
+              <h3>
+                Name: <span>{user.username}</span> <br />
+                Email: <span>{user.email}</span> <br />
+              </h3>
+            </div>
+
+            <Formik
+              initialValues={{
+                username: user.username,
+                email: user.email,
+                imageURL: user.imageURL
+              }}
+              validate={values => profilePageValidator(values)}
+              onSubmit={async (
+                { username, email, imageURL },
+                { setSubmitting }
+              ) => {
+                if (imageURL.length === 0) {
```

Comment:
This line was created to set the imageURL to null if the string is empty. I dont want to fix it from the server side.(A new PR will be made to that effect on another Story). My test for empty string wasnt created properly.

Reply:
Okay. fair enough.

Reply:
However, the app currently errors as is when I try to submit. To this end, a temporary fix might be to update condition to `if(!imageURL)`